packages/protocol/contracts/rrp/authorizers/Api3TokenLock.sol

```diff
@@ -0,0 +1,305 @@
+        airnodeRequester.sponsorUnlockTime[msg.sender] = expirationTime;
+        airnodeRequester.whitelistCount++;
+
+        assert(
+            IApi3Token(api3Token).transferFrom(
+                msg.sender,
+                address(this),
+                lockAmount
+            )
+        );
+
+        if (airnodeRequester.whitelistCount == 1) {
+            IApi3RequesterRrpAuthorizer(api3RequesterRrpAuthorizer)
+                .setWhitelistExpiration(
+                    bytes32(abi.encode(_airnode)),
+                    _requesterAddress,
+                    type(uint64).max
+                );
+        }
+
+        emit Lock(_airnode, _requesterAddress, msg.sender, lockAmount);
+    }
+
+    /// @notice Unlocks the API3 tokens for a given airnode-requester pair.
+    /// Airnode-requester pair will be unauthorized if no token lock for the pair is found.
+    /// @dev Checks whether the user (msg.sender) has already locked anything,
+    /// if the locked period has expired and if the requester address is blocked
+    /// Transfers to msg.sender the locked amount.
+    /// @param _airnode The airnode address
+    /// @param _requesterAddress The requester address
+    function unlock(address _airnode, address _requesterAddress)
+        external
+        override
+        isNotBlocked(_airnode, _requesterAddress)
+    {
+        AirnodeRequester
+            storage airnodeRequester = airnodeToRequesterToTokenLocks[_airnode][
+                _requesterAddress
+            ];
+        require(
+            airnodeRequester.sponsorLockAmount[msg.sender] != 0,
+            ERROR_NOT_LOCKED
+        );
+        require(
+            airnodeRequester.sponsorUnlockTime[msg.sender] != 0,
+            ERROR_LOCK_PERIOD_ZERO
+        );
+        require(
+            airnodeRequester.sponsorUnlockTime[msg.sender] <= block.timestamp,
+            ERROR_LOCK_PERIOD_NOT_EXPIRED
+        );
+        uint256 amount = airnodeRequester.sponsorLockAmount[msg.sender];
+
+        airnodeRequester.sponsorLockAmount[msg.sender] = 0;
+        airnodeRequester.sponsorUnlockTime[msg.sender] = 0;
+        airnodeRequester.whitelistCount--;
+
+        if (airnodeRequester.whitelistCount == 0) {
+            IApi3RequesterRrpAuthorizer(api3RequesterRrpAuthorizer)
+                .setWhitelistExpiration(
+                    bytes32(abi.encode(_airnode)),
+                    _requesterAddress,
+                    0
+                );
+        }
+
+        assert(IApi3Token(api3Token).transfer(msg.sender, amount));
+
+        emit Unlock(_airnode, _requesterAddress, msg.sender, amount);
+    }
+
+    /// @notice User calls this when the lock amount has been decreased and wants
+    /// to withdraw the redundantly locked tokens
+    /// @param _airnode The airnode address
+    /// @param _requesterAddress The requester address
+    function withdrawExcess(address _airnode, address _requesterAddress)
```

Comment:
Remainder maybe? Again, I'm not good at naming things but excess sounds a litte bit weird to me 🙃

Reply:
> As adjectives the difference between remainder and excess is that remainder is remaining while excess is more than is normal, necessary or specified.

I think I'll keep excess.